cytoscape-edgehandles.js

```diff
@@ -619,12 +765,21 @@ SOFTWARE.
               return; // nothing to do :(
             }
 
+            const hit = hitTest(targets, { x: mx, y: my });
+
             // just remove preview class if we already have the edges
             if( !src && !tgt ) {
               if( !preview && options().preview ) {
                 added = cy.elements( '.edgehandles-preview' ).removeClass( 'edgehandles-preview' );
-                options().complete( source, targets, added );
-                source.trigger( 'cyedgehandles.complete' );
+
+                if ( !hit ) {
+                  added.remove();
```

Comment:
In original code, an edge is created when drag is released on anywhere on node. 
But in my application, I realized that it was helpful to make it more specific on release on a handle.   
Though, if not-restricting to a handle is more popular, I can remove this.